src/server/db/migrations/2020-11-13_0.3.0_to_0.4.0/migrate_0.3.0_to_0.4.0.js

```diff
@@ -0,0 +1,414 @@
+    "keywordString",            // 5.e
+    "languageFromSelect",       // 5.f
+    "languageFromInput",        // 5.g
+    "requestedTranslations",    // 5.h
+    "summary",                  // 5.i
+    "thumbnailAltText",         // 5.j
+    "thumbnailUrl",             // 5.k
+    "translationOf"             // 5.l
+];
+
+/**
+ * Create a set of options for data loader and a function to retreive them.
+ * The options are based on the command line parameters and a set of database
+ * constants
+ *
+ * @param {Array} processArgv - The command line arguments.
+ *
+ * @return {Object} - The options.
+ */
+sjrk.storyUpdate.initOptions = function (processArgv) {
+    var options = {};
+    options.couchDbUrl = processArgv[2];
+    options.maxDocsInBatchPerRequest = Number(processArgv[3]) || sjrk.storyUpdate.defaultMaxDocsInBatchPerRequest;
+
+    options.storiesUrl = options.couchDbUrl + "/_all_docs?include_docs=true&limit=" + options.maxDocsInBatchPerRequest;
+
+    // Set up database specific options
+    options.parsedCouchDbUrl = url.parse(options.couchDbUrl);
+    options.postOptions = {
+        hostname: options.parsedCouchDbUrl.hostname,
+        port: options.parsedCouchDbUrl.port,
+        path: "/stories/_bulk_docs",
+        auth: options.parsedCouchDbUrl.auth,
+        method: "POST",
+        headers: {
+            "Accept": "application/json",
+            "Content-Length": 0, // this value is filled in at runtime
+            "Content-Type": "application/json"
+        }
+    };
+    fluid.log("COUCHDB_URL: '" +
+        options.parsedCouchDbUrl.protocol + "//" +
+        options.parsedCouchDbUrl.hostname + ":" +
+        options.parsedCouchDbUrl.port +
+        options.parsedCouchDbUrl.pathname + "'"
+    );
+    return options;
+};
+
+/**
+ * Create the step that retrieves stories from the database
+ *
+ * @param {Object} options - options for the query
+ *
+ * @return {Promise} - A promise that resolves retrieving the stories.
+ */
+sjrk.storyUpdate.retrieveStories = function (options) {
+    var details = {
+        requestUrl: options.storiesUrl,
+        requestErrMsg: "Error retrieving stories from the database: ",
+        responseDataHandler: sjrk.storyUpdate.updateStoriesData,
+        responseErrMsg: "Error retrieving stories from database: "
+    };
+    return gpii.dbRequest.configureStep(details, options);
+};
+
+/**
+ * Given all the documents from the database, update their data according to the new data model
+ *
+ * @param {String} responseString - the response from the request to get all documents
+ * @param {Object} options - Where to store the to-be-updated documents
+ *
+ * @return {Array} - The updated documents in the new data structure with the new schema version.
+ */
+sjrk.storyUpdate.updateStoriesData = function (responseString, options) {
+    var allDocs = JSON.parse(responseString);
+    var updatedDocs = [];
+    options.totalNumOfDocs = allDocs.total_rows;
+
+    if (allDocs.rows) {
```

Comment:
This test is unnecessary, fluid.each is a no-op on missing arguments. However the intention is clearer if this is a fluid.transform of allDocs.rows (which will also return non-values unchanged)